Angular2-Starter-Project.md

```diff
@@ -0,0 +1,232 @@
+
+Let's name our file: app/increment-clicker.component.ts and set it up.
+```
+// app/increment-clicker.component.ts
+import { Component } from '@angular/core'; 
+
+@Component({
+    selector: 'increment-clicker',
+    template: `
+      <div className="counter">
+        <h1>{{curClicks}} clicks</h1>
+        <button type="button" (click)="incrementClicks()">Increment</button>
+      </div>
+    `,
+    styles: [`
+        .counter {
+          width: 100%;
+          margin: auto;
+          background: darkgreen;
+          border-radius: 5px;
+          color: white;
+          padding: 20px;
+          text-align: center;
+        }
+        .counter h1 {
+          margin: 0;
+          padding: 20px;
+          font-size: 36px;
+        }
+        .counter button {
+          background: #f1c40f;
+          border: 0;
+          box-shadow: 0px 5px 0px #927608;
+          padding: 20px;
+          width: 100%;
+          outline: none;
+          border-radius: 5px;
+          color: darkgreen;
+          font-weight: bold;
+        }
+        
+        .counter button:hover {
+          background: #a9890a;
+          cursor: pointer;
+        }
+    `]
+})
+
+export class IncrementClicker {
+  curClicks = 0;
+  
+  incrementClicks() {
+    this.curClicks++;
+  }  
+}
+```
+Our component structure is similar to our previous one, but we've added a bit of functionality here. You will notice our class is no longer empty. We've added a variable to track the clicks as well as a function to increment them when the user clicks the button.
+
+You can display variables from your class in your template using double brackets such as: {{variableName}}
+
+We bind the button click to our class function using Angulars Event Bindings.
+```
+<button type="button" (click)="incrementClicks()">Increment</button>
+```
+
+Notice that we've also added some styling to our components. This way everything is self-contained in this module. If you include it elsewhere you won't have to worry about adding the style rules to a master css file, etc.
+
+Now we just need to update our AppComponent to include our new component:
+```
+// app/app.component.ts
+
+import { Component } from '@angular/core';
+
+/** Nested Component */ 
+import { IncrementClicker } from 'app/increment-clicker.component';
+
+@Component({
+  selector: 'my-app',
+  directives: [IncrementClicker],
+  template: `
+    <h1>Free Code Camp Rocks!</h1>
+    <increment-clicker></increment-clicker>
+  `,
+  styles: [`
+    h1 { color: darkgreen; font-family: Arial, Helvetica, sans-serif;}
+  `]  
+})
+
+export class AppComponent { }
+```
+To pull in our new IncrementClicker component we need to import it by including the classname as well as the location of the typescript file as we did on this line:
+```
+import { IncrementClicker } from 'app/increment-clicker.component';
+```
+We then add the directives to our AppComponent using:
```

Comment:
We then add the `IncrementClicker` *component* as a directive to the `AppComponent` module using:

```md
We then add the `IncrementClicker` *component* as a directive to the `AppComponent` module using:
```